oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/listener/vservice/VirtualMQProcessor.java

```diff
@@ -51,17 +52,25 @@ public void prepareVSIfNecessary(final SpanObject span, final SegmentObject segm
             return;
         }
         MQTags mqTags = collectTags(span.getTagsList());
-        if (StringUtil.isBlank(mqTags.broker)) {
-            return;
+        final MQOperation mqOperation;
+        final String serviceName;
+        if (span.getSpanType() == SpanType.Entry) {
+            mqOperation = MQOperation.Consume;
+            final String peer = span.getRefsList()
+                                    .stream()
+                                    .findFirst()
+                                    .map(SegmentReference::getNetworkAddressUsedAtPeer)
+                                    .orElse(null);
+            serviceName = namingControl.formatServiceName(peer);
+        } else {
```

Comment:
Should we have an `else if` here?

Reply:
Yes ， some plugin would create localSpan .  we only need exit entry span .